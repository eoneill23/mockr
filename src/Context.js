import { createContext } from 'react';

export const QuestionsContext = createContext([]);
export const InterviewsContext = createContext([]);
export const UserContext = createContext('');