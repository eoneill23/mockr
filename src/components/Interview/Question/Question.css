.shadow {box-shadow: 0 16px 24px 2px rgba(0,0,0,0.15), 0 6px 30px 5px rgba(0,0,0,0.12), 0 8px 10px -7px rgba(0,0,0,0.2);}

.box-fix {
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}

.card {
  background: #48a5a0;
  color: white;
  padding: 2rem;

  position: absolute;
  top: 2rem; bottom: 2rem;
  border-radius: 0.5rem;
  transition: transform 0.3s linear, opacity 0.3s linear;

  left: 5%; right: 5%; /* screen < 800px */
}

.card-question {
  font-size: 3rem;
  font-weight: 300;
  width: 90%;
  max-height: 35%;
  margin: 0;
  margin-top: 2rem;
}

.card-position {
  font-size: 4rem;
  font-weight: 400;
  margin: 0;
  position: absolute;
  top: 3rem;
  right: 3rem;
}

.header-notes {
  color: white;
  font-size: 2rem;
  font-weight: 300;
  position: absolute;
  bottom: calc(30% + 6rem); left: 2rem;
}

.card-response {
  resize: none;
  font-size: 1.5rem;
  color: white;
  background: transparent;
  border-top: 0.2rem solid white;
  border-bottom: 0.2rem solid white;
  border-left: none; border-right: none;

  position: absolute;
  bottom: 2rem; left: 2rem;
  width: calc(100% - 20rem);
  height: 40%;
}
.card-response:hover {background: rgba(255,255,255,0.1);}
.card-response:focus {outline: none; background: rgba(50,50,50,0.2);}

.card-score {
  font-size: 1.2rem;

  position: absolute;
  bottom: 10rem; right: 1.5rem;
  height: calc(40% - 8rem);
}
.card-score input {
  background: transparent;
}

.card-next-btn {
  position: absolute;
  left: calc(100% - 8rem);
  bottom: 3rem;
  width: 5rem; height: 5rem;
  background: #f03750;
  border: 0;
  border-radius: 5rem;
  color: #9b0528;
  font-size: 4.5rem;
  line-height: 2rem;
  margin: 0; padding: 0;
  cursor: pointer;

  display: none;
}
.card-next-btn:focus {outline: 0;}

@media only screen and (min-width: 800px) {
  .card {left: 20%; right: 20%;}
  .card-response {width: calc(100% - 13rem);}
  .card-next-btn {display: block;}
}
@media only screen and (min-width: 1400px) {
  .card {left: 30%; right: 30%;}
}




.card-focused {
  z-index: 0;
}

.card-first {
  z-index: -1;
  transform: translate(1rem, 0.5rem);
}

.card-second {
  z-index: -2;
  transform: translate(2rem, 1rem);
}

.card-third {
  z-index: -3;
  transform: translate(3rem, 1.5rem);
}

.card-hidden {
  z-index: -4;
  opacity: 0;
  transform: translate(4rem, 2rem);
}

.card-finished {
  z-index: 1;
  transform: translate(-100vw, 0);
}

