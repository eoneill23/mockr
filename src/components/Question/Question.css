.shadow {box-shadow: 0 16px 24px 2px rgba(0,0,0,0.15), 0 6px 30px 5px rgba(0,0,0,0.12), 0 8px 10px -7px rgba(0,0,0,0.2);}

.card {
  background: #48a5a0;
  color: white;
  padding: 2rem;

  position: fixed;
  top: 6rem; bottom: 2rem;
  left: 20%; right: 20%;
  border-radius: 0.5rem;
  transition: transform 0.3s linear, opacity 0.3s linear;
}

.card-question {
  font-size: 3rem;
  font-weight: 300;
  width: 90%;
  max-height: 40%;
}

.card-position {
  font-size: 4rem;
  font-weight: 400;
  margin: 0;
  position: absolute;
  top: 3rem;
  right: 3rem;
}



.card-focused {
  z-index: 0;
}

.card-first {
  z-index: -1;
  transform: translate(1rem, 0.5rem);
}

.card-second {
  z-index: -2;
  transform: translate(2rem, 1rem);
}

.card-third {
  z-index: -3;
  transform: translate(3rem, 1.5rem);
}

.card-hidden {
  z-index: -4;
  opacity: 0;
  transform: translate(4rem, 2rem);
}

.card-finished {
  z-index: 1;
  transform: translate(-100vw, 0);
}

