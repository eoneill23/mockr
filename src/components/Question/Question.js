import React from 'react';

export const Question = ({ body, notes, id, showDetails, detailed }) => {

  const isDetailed = () => {if (detailed) {return ' shown'} else {return ''}}
  return (
    <section className='question-card' onClick={e => showDetails(id)}>
      <h3>{body}</h3>
      <div className={'details' + isDetailed()}>
        <p>{notes}</p>
      </div>
    </section>
  )
}

export default Question;
