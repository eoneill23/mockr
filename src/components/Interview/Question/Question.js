import React from 'react';
import './Question.css';

export const Question = props => {
  const whereAmI = cur => {
    let pos = parseInt(props.pos);
    if (pos < cur) {
      return 'card-finished';
    } else if (pos === cur) {
      return 'card-focused';
    } else {
      if (pos === (cur + 1)) {
        return 'card-first';
      } else if (pos === (cur + 2)) {
        return 'card-second';
      } else if (pos === (cur + 3)) {
        return 'card-third';
      } else {
        return 'card-hidden';
      }
    }
  }

  return (
    <div className={'card shadow ' + whereAmI(props.cur)}>
      <h1 className='card-question'>Q: {props.question}</h1>
      <h1 className='card-position'>{props.pos}</h1>

      <form id={'card-response-' + props.id}>
        <h3 className='header-notes'>Notes:</h3>
        <textarea name='response' form={'card-response-' + props.pos} className='box-fix card-response' onChange={event => props.fs.note(props.id, event.target.value)}></textarea>
      </form>
      <button className='card-next-btn shadow' onClick={props.fs.next} value='ðŸ š'/>
    </div>
  );
}

export default Question;
