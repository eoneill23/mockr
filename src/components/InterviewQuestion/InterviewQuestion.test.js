import React, { useMemo } from 'react';
import { render } from '@testing-library/react';
import Question from './InterviewQuestion';
import { MockedProvider } from '@apollo/react-testing';
import { configure, mount } from 'enzyme';
import { UserContext } from '../../Context';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Question', () => {
  let wrapper, mockUser, mockStudent;

  beforeEach(() => {
    mockUser = {
      user: {
        name: 'Eric',
        role: 1,
        currentInterview: {
          id: 1
        }
      },
      setUser: jest.fn()
    }
    mockStudent = {
      firstName: 'Eric',
      lastName: 'O\'Neill'
    }
    wrapper = mount(
      <MockedProvider addTypeName={false}>
        <UserContext.Provider value={mockUser}>
          <Question student={mockStudent} />
        </UserContext.Provider>
      </MockedProvider>
    );
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});