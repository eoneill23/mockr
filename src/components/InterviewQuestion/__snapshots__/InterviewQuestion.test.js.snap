// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should match the snapshot with all data passed in correctly 1`] = `
<MockedProvider
  addTypeName={false}
  addTypename={true}
>
  <ApolloProvider
    client={
      ApolloClient {
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "clearStoreCallbacks": Array [],
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": MockLink {
          "addTypename": true,
          "mockedResponsesByKey": Object {},
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "client": [Circular],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "queryManager": QueryManager {
          "assumeImmutableResults": false,
          "clientAwareness": Object {
            "name": undefined,
            "version": undefined,
          },
          "dataStore": DataStore {
            "cache": InMemoryCache {
              "addTypename": true,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
          },
          "fetchQueryRejectFns": Map {},
          "idCounter": 1,
          "inFlightLinkObservables": Map {},
          "link": MockLink {
            "addTypename": true,
            "mockedResponsesByKey": Object {},
          },
          "localState": LocalState {
            "cache": InMemoryCache {
              "addTypename": true,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
            "client": [Circular],
          },
          "mutationStore": MutationStore {
            "store": Object {},
          },
          "onBroadcast": [Function],
          "pollingInfoByQueryId": Map {},
          "queries": Map {},
          "queryDeduplication": true,
          "queryStore": QueryStore {
            "store": Object {},
          },
          "ssrMode": false,
          "transformCache": WeakMap {},
        },
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "typeDefs": undefined,
        "version": "2.6.4",
        "watchQuery": [Function],
      }
    }
  >
    <Question
      student={
        Object {
          "firstName": "Eric",
          "lastName": "O'Neill",
        }
      }
    >
      <div
        className="interview-question-card shadow interview-card-hidden"
      >
        <h1
          className="interview-card-question"
        >
          Q: 
        </h1>
        <h1
          className="interview-card-position"
        />
        <h3
          className="header-notes"
          id="notes-h3"
        >
          Notes:
        </h3>
        <textarea
          className="box-fix interview-card-response"
          form="interview-card-response-undefined"
          name="response"
          onChange={[Function]}
        />
        <form
          className="interview-card-score"
          id="score-form-undefined"
          onChange={[Function]}
        >
          <label>
            <input
              name="score"
              type="radio"
              value="1"
            />
            Unsatisfactory
            <br />
          </label>
          <label>
            <input
              name="score"
              type="radio"
              value="2"
            />
            Needs Work
            <br />
          </label>
          <label>
            <input
              name="score"
              type="radio"
              value="3"
            />
            Good
            <br />
          </label>
          <label>
            <input
              name="score"
              type="radio"
              value="4"
            />
            Exceptional
            <br />
          </label>
        </form>
        <button
          className="interview-card-next-btn shadow"
          onClick={[Function]}
        >
          Next
        </button>
        <button
          className="interview-card-skip-btn"
          onClick={[Function]}
        >
          Skip
        </button>
      </div>
    </Question>
  </ApolloProvider>
</MockedProvider>
`;
